src/hints.rs

```diff
@@ -138,6 +141,7 @@ impl<'a> From<&'a NotificationHint> for MessageItem {
             &NotificationHint::Y(value)                 => (Y              .to_owned(), MessageItem::Int32(value)),
             &NotificationHint::Urgency(value)           => (URGENCY        .to_owned(), MessageItem::Byte(value as u8)),
             &NotificationHint::Custom(ref key, ref val) => (key            .to_owned(), MessageItem::Str(val.to_owned ())),
+            &NotificationHint::CustomInt(ref key, val)  => (key            .to_owned(), MessageItem::Int32(val)),
```

Comment:
Right, I just played around with it for a few minutes, but it may be a little more work than I anticipated, would you like to give it a try?